src/sqlitedb.cpp

```diff
@@ -279,30 +286,94 @@ bool DBBrowserDB::tryEncryptionSettings(const QString& filename, bool* encrypted
+                    sqlite3_close(dbHandle);
+                    if(sqlite3_open_v2(filePath.toUtf8(), &dbHandle, SQLITE_OPEN_READONLY, nullptr) != SQLITE_OK)
+                    {
+                        return false;
+                    }
+
+                    // Set the key
+                    sqlite3_exec(dbHandle, QString("PRAGMA key = %1").arg(cipherSettings->getPassword()).toUtf8(), nullptr, nullptr, nullptr);
+
+                    // Set the page size if it differs from the default value
+                    if(cipherSettings->getPageSize() != CipherSettings::defaultPageSize)
+                        sqlite3_exec(dbHandle, QString("PRAGMA cipher_page_size = %1;").arg(cipherSettings->getPageSize()).toUtf8(), nullptr, nullptr, nullptr);
+
+                    *encrypted = true;
+
+                    // Skip the CipherDialog prompt for now to test if the saved password was correct
+                    continue;
+                }
+            }
+
+            CipherDialog *cipherDialog = new CipherDialog(nullptr, false);
+            if(cipherDialog->exec())
             {
                 // Close and reopen database first to be in a clean state after the failed read attempt from above
                 sqlite3_close(dbHandle);
-                if(sqlite3_open_v2(filename.toUtf8(), &dbHandle, SQLITE_OPEN_READONLY, NULL) != SQLITE_OK)
+                if(sqlite3_open_v2(filePath.toUtf8(), &dbHandle, SQLITE_OPEN_READONLY, nullptr) != SQLITE_OK)
                 {
                     delete cipherSettings;
                     cipherSettings = nullptr;
                     return false;
                 }
 
-                // Set key and, if it differs from the default value, the page size
-                sqlite3_exec(dbHandle, QString("PRAGMA key = %1").arg(cipherSettings->password()).toUtf8(), NULL, NULL, NULL);
-                if(cipherSettings->pageSize() != 1024)
-                    sqlite3_exec(dbHandle, QString("PRAGMA cipher_page_size = %1;").arg(cipherSettings->pageSize()).toUtf8(), NULL, NULL, NULL);
+                if (cipherSettings)
+                    delete cipherSettings;
+
+                cipherSettings = new CipherSettings(cipherDialog->getCipherSettings());
+
+                // Set the key
+                sqlite3_exec(dbHandle, QString("PRAGMA key = %1").arg(cipherSettings->getPassword()).toUtf8(), nullptr, nullptr, nullptr);
+
+                // Set the page size if it differs from the default value
+                if(cipherSettings->getPageSize() != CipherSettings::defaultPageSize)
+                    sqlite3_exec(dbHandle, QString("PRAGMA cipher_page_size = %1;").arg(cipherSettings->getPageSize()).toUtf8(), nullptr, nullptr, nullptr);
 
                 *encrypted = true;
             } else {
                 sqlite3_close(dbHandle);
                 *encrypted = false;
-                delete cipherSettings;
-                cipherSettings = nullptr;
+                if (cipherSettings)
```

Comment:
Same here. No need for the ```if``` here, the old code did the same but was a bit shorter and easier to read.